src/Microsoft.ML.AutoML/API/ExperimentBase.cs

```diff
@@ -102,16 +102,16 @@ public abstract class ExperimentBase<TMetrics, TExperimentSettings>
             const int crossValRowCountThreshold = 15000;
 
             var rowCount = DatasetDimensionsUtil.CountRows(trainData, crossValRowCountThreshold);
-
-            if (rowCount < crossValRowCountThreshold)
+            bool isImageClassificationTask = columnInformation.ImagePathColumnNames.Count != 1;
+            if (!isImageClassificationTask && rowCount < crossValRowCountThreshold)
```

Comment:
As a continuation of https://github.com/dotnet/machinelearning/pull/4384#discussion_r338960267

---

Shuffling is not required for CV to work properly. And CV is required when your number of rows is small, to get a useful measure of how good your model performs.

I think you're visualizing the CV folds as created as `folds =  [ rows 0...99, rows 100...199, rows 200..299 ]`. Instead of this method, the rows are split pseudorandomly.

CV splits the dataset with each row going into one of K folds pseudorandomly. Then each of the K folds are chosen one at a time as the test dataset, and the other K-1 folds are used as the training set. 

The cats and dogs in your example will be roughly equally split among the CV folds. That said, you may want a shuffle as part of your trainer, to lessen overfitting.

```suggestion
            if (rowCount < crossValRowCountThreshold)
```

Reply:
Hi Justin, 

I don't want our ImageClassificationTrainer to be used on CV. For the initial release it will be disabled until we figured out there is value in doing CV. Contrary to what you had suggested, using train-test on smaller datasets is actually give great accuracy, please see my tests, it works on trainset of 50 images. 

---
In reply to: [339329519](https://github.com/dotnet/machinelearning/pull/4395#discussion_r339329519) [](ancestors = 339329519)

Reply:
The current transition point for TrainTest vs. CV was chosen based on benchmarking on over 100 datasets in the AutoML team. We are very confident on its importance. 

Disabling CV gives you a 10x speed up, but makes the accuracy metric too noisy on a dataset with only 50 images for a unit test. At a seeded-random 80/20 split only 9 images are the validation split. Hence the accuracy metric will jump by 11% for each image being right or wrong. 

This gives a random value for the output accuracy metric. It's not  a _great_ accuracy, instead it's roughly _unknown_.

We're trying to measure the value of a weighted coin (percent of correct predictions). The actual model accuracy is perhaps 0.7890, but given only 9 datapoint in the unit test's measurement set we will know the accuracy only with enormous error bars.

The point of using TrainTest mode or CV mode is to estimate how well the user's model will do in production. Currently your model's accuracy is somewhere between 55% to 91% (0.95 confidence interval for a Bernoulli distribution w/ 8 of 9 correct predictions)<sup>[[1]](http://epitools.ausvet.com.au/content.php?page=CIProportion&SampleSize=9&Positive=8&Conf=0.95&method=6&Digits=4)</sup>. Bringing in CV mode makes your error bars significantly smaller, which is its utility.

CV is how you get a useful metric on a small dataset. Currently the code returns a non-useful metric for small datasets.

Reply:
Thanks, @justinormont. Another topic for offline discussion :)